pkg/version/version.go

```diff
@@ -191,3 +197,85 @@ func IsReleaseVersion() bool {
 		GetMainVersion() != "" &&
 		!strings.Contains(GetMainVersion(), "-")
 }
+
+// IsPriorToScannerV4 returns true if version represents a version of ACS from prior to the
+// introduction of Scanner V4. Will return an error if cannot determine result.
+func IsPriorToScannerV4(version string) (bool, error) {
+	parsed, err := parseVersion(version)
+	if err != nil {
+		return false, err
+	}
+
+	x := parsed.MarketingMajor
+	y := parsed.EngRelease
+	z := parsed.PatchLevel
+
+	if x < 4 {
+		return true, nil
+	}
+
+	if x > 4 {
+		return false, nil
+	}
+
+	// x == 4
+	if y < 3 || (y == 3 && (z == "" || z != "x")) {
+		return true, nil
+	}
+
+	return false, nil
```

Comment:
Since we know `x == 4`, we can just do this

```suggestion
	// x == 4
	// Scanner V4 was introduced in 4.4
	// TODO(ticket here?): Once 4.4 is release, remove extra z checks
	return y < 3 || (y == 3 && (z == "" || z != "x")), nil
```

Reply:
Don't believe the TODO is necessary and the `z` logic can be kept - If the current git ref that is checked out happens to be older, this logic will still be accurate.  ie:

```
$ git describe
4.3.x-1100-g2f80ec8122

$ git checkout 4.2.x
$ git describe
4.2.x-0-g4fde452451
```

Reply:
https://github.com/stackrox/stackrox/pull/9803/commits/b85f3b05bd969178d5518344cf80914d21e77271